cmd/delete.go

```diff
@@ -78,28 +87,29 @@ func runDelete(cmd *cobra.Command, args []string, newClient ClientFactory) (err
 			Name: args[0],
 		}
 	} else {
-		function, err = fn.NewFunction(config.Path)
+		function, err = fn.NewFunction(cfg.Path)
 		if err != nil {
 			return
 		}
 
 		// Check if the function has been initialized
 		if !function.Initialized() {
-			return fmt.Errorf("the given path '%v' does not contain an initialized function", config.Path)
+			return fmt.Errorf("the given path '%v' does not contain an initialized function", cfg.Path)
+		}
+
+		// If not provided, use the function's extant namespace
+		if !cmd.Flags().Changed("namespace") {
```

Comment:
Why is this defaulting done in the else branch when before it was done after the else?